BlobStorage/Advanced.cs

```diff
@@ -1277,30 +1151,33 @@ private static async Task ListBlobsHierarchicalListingAsync(CloudBlobContainer c
         /// </summary>
         /// <param name="container">The container.</param>
         /// <returns>A Task object.</returns>
-        private static async Task<CloudBlockBlob> CreateRandomlyNamedBlockBlobAsync(CloudBlobContainer container)
+        private static async Task<BlobClient> CreateRandomlyNamedBlockBlobAsync(BlobContainerClient container)
         {
             // Get a reference to a blob that does not yet exist.
             // The GetBlockBlobReference method does not make a request to the service, but only creates the object in memory.
             string blobName = BlobPrefix + Guid.NewGuid();
-            CloudBlockBlob blob = container.GetBlockBlobReference(blobName);
+            BlobClient blob = container.GetBlobClient(blobName);
 
             // For the purposes of the sample, check to see whether the blob exists.
-            Console.WriteLine("Blob {0} exists? {1}", blobName, await blob.ExistsAsync());
+            Console.WriteLine("Blob {0} exists? {1}", blobName, blob.ExistsAsync().GetAwaiter().GetResult().Value);
 
             try
             {
-                // Writing to the blob creates it on the service.
-                await blob.UploadTextAsync(string.Format("This is a blob named {0}", blobName));
+                using (var stream = new MemoryStream(Encoding.UTF8.GetBytes(string.Format("This is a blob named {0}", blobName))))
+                {
+                    // Writing to the blob creates it on the service.
+                    await blob.UploadAsync(stream);
+                }
```

Comment:
Just do `await blob.UploadContent(BinaryData.FromString($"This is a blob named {blobName}"))` instead